src/EventStore.ClientAPI/ClientOperations/SubscriptionOperation.cs

```diff
@@ -257,7 +259,10 @@ protected void EventAppeared(TE e)
         private void ExecuteActionAsync(Func<Task> action)
         {
             _actionQueue.Enqueue(action);
-            if (_actionQueue.Count > _maxQueueSize) DropSubscription(SubscriptionDropReason.UserInitiated, new Exception("client buffer too big"));
+            
+            if (_actionQueue.Count > _maxQueueSize) 
```

Comment:
This change make sense as you are redefining the exception in order to handle it properly. I don't understand all the other changes related to the maxQueueSize setting and the default value.

Reply:
Yes, I can. But I'd like to know what is the reason for the `_maxQueueSize` default value to be hardcoded instead of using the setting?

Reply:
That is an internal setting that has been decided to be set 2000 as an appropriate value. 

The setting MaxQueueSize is already used in the EventStoreNodeConnection class to limit the maximum number of outstanding items allowed in the queue

https://github.com/EventStore/EventStore/blob/7e98bd4206d5b0aec0cbc24d12002dfe195643fe/src/EventStore.ClientAPI/Internal/EventStoreNodeConnection.cs#L242

I like the improvement of changing the SubscriptionDropReason but not the other changes that extend the use of MaxQueueSize setting in the EventStoreConnectionLogicHandler class.

Reply:
Ok, sounds reasonable. I will remove other changes.

Reply:
Hit this bug myself in a production system today. Since @alexeyzimarev didn't get round to making these changes yet i've done so in #1747.